src/robotFactory.js

```diff
@@ -1,10 +1,72 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    this.coords = {
+      x: coords.x || 0,
+      y: coords.y || 0,
+    };
```

Comment:
```suggestion
    this.coords = {
      x,
      y,
    };
```
You should handle possible undefined coords parameter in the BaseRobot constructor. If the coords parameter is not passed to the BaseRobot constructor, trying to access coords.x or coords.y will result in a TypeError.